core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -815,6 +815,21 @@ public Optional<TableLayout> getNewTableLayout(Session session, String catalogNa
                 .map(layout -> new TableLayout(catalogHandle, transactionHandle, layout));
     }
 
+    @Override
+    public ConnectorTableMetadata getNewTableMetadata(Session session, String catalogName, QualifiedObjectName table, List<ColumnMetadata> columns, Map<String, Object> properties, Optional<String> comment)
+    {
+        CatalogMetadata catalogMetadata = getRequiredCatalogMetadata(session, catalogName);
+        CatalogHandle catalogHandle = catalogMetadata.getCatalogHandle(session, table);
+        ConnectorMetadata metadata = catalogMetadata.getMetadataFor(session, catalogHandle);
+        ConnectorSession connectorSession = session.toConnectorSession(catalogHandle);
+        return metadata.getNewTableMetadata(
```

Comment:
should we check that number of columns returned by `metadata.getNewTableMetadata` matches `columns` from arguments?